src/cargo/util/lints.rs

```diff
@@ -144,7 +144,7 @@ impl From<TomlLintLevel> for LintLevel {
 /// [RFC #3491]: https://rust-lang.github.io/rfcs/3491-remove-implicit-features.html
 const IMPLICIT_FEATURES: Lint = Lint {
     name: "implicit_features",
-    desc: "warn about the use of unstable features",
+    desc: "implicit features for optional dependencies is deprecated and will be unavailable in the 2024 edition",
```

Comment:
nit: I would have had this in the commit where this message is used

(this is more for future reference)